### PR TITLE
Enable `eslint-config-prettier`

### PR DESCRIPTION
**Pull request by basil**
Tested with:

- `npm install`
- `npm test`
- `npm run lint`
- `npm run prettier`
- `npm run build`

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -4,18 +4,19 @@ module.exports = {
     node: true,
   },
   extends: [
-    'eslint:recommended',
     'plugin:vue/vue3-essential',
-    '@vue/eslint-config-typescript',
+    '@vue/eslint-config-typescript/recommended',
+    '@vue/eslint-config-prettier',
```
Comment (basil) [inline .eslintrc.js:9]: Following [the documentation](https://github.com/vuejs/eslint-config-typescript#vueeslint-config-typescriptrecommended).

File: .eslintrc.js [hunk 2]
```diff
@@ -4,18 +4,19 @@ module.exports = {
...
   ],
   rules: {
     'no-console':
       process.env.NODE_ENV === 'production'
         ? ['error', { allow: ['warn', 'error'] }]
         : 'off',
     'no-debugger': process.env.NODE_ENV === 'production' ? 'error' : 'off',
-    'no-unused-vars': 'off',
-    '@typescript-eslint/no-unused-vars': 'error',
```
Comment (basil) [inline .eslintrc.js:18]: These were no longer needed. I verified this by intentionally introducing an unused variable and verifying that a `@typescript-eslint/no-unused-vars` error was thrown.

File: .eslintrc.js [hunk 3]
```diff
@@ -4,18 +4,19 @@ module.exports = {
...
+    '@typescript-eslint/no-explicit-any': 'off',
+    '@typescript-eslint/no-inferrable-types': 'off',
+    '@typescript-eslint/no-non-null-assertion': 'off',
```
Comment (basil) [inline .eslintrc.js:19]: These three errors are pervasive in this codebase. Disabling the checks is no worse than the existing status quo, but it allows enforcement of the other checks in the `@vue/eslint-config-typescript/recommended` ruleset, which is an improvement over the status quo.

File: package.json [hunk 4]
```diff
@@ -54,13 +54,15 @@
...
     "@vitejs/plugin-vue": "^4.2.3",
+    "@vue/eslint-config-prettier": "^7.1.0",
     "@vue/eslint-config-typescript": "^11.0.3",
     "electron": "^24.6.2",
     "electron-builder": "^24.4.0",
     "electron-devtools-installer": "^3.2.0",
     "electron-icon-builder": "^2.0.1",
     "electron-updater": "^6.1.1",
     "eslint": "^8.44.0",
+    "eslint-plugin-prettier": "^5.0.0-alpha.2",
```
Comment (basil) [inline package.json:65]: Alpha version is required due to https://github.com/prettier/eslint-plugin-prettier/issues/562.

File: src/components/ImageBox.vue [hunk 5]
```diff
@@ -24,7 +24,6 @@
 import { Component, Prop, Vue } from 'vue-facing-decorator';
 import { ImageBoxElement } from '@/models/Element';
 import ContentEditable from '@/components/ContentEditable.vue';
-// @ts-ignore
```
Comment (basil) [inline src/components/ImageBox.vue:27]: No longer needed apparently, and raises a warning if present.

File: src/components/ToolbarMain.vue [hunk 6]
```diff
@@ -243,8 +243,8 @@ export default class ToolbarMain extends Vue {
   @Prop() entryMode!: EntryMode;
   @Prop() zoom!: number;
   @Prop() zoomToFit!: boolean;
-  @Prop() currentPageNumber!: Number;
-  @Prop() pageCount!: Number;
+  @Prop() currentPageNumber!: number;
+  @Prop() pageCount!: number;
```
Comment (basil) [inline src/components/ToolbarMain.vue:247]:
```
  246:31  error  Don't use `Number` as a type. Use number instead  @typescript-eslint/ban-types
  247:23  error  Don't use `Number` as a type. Use number instead  @typescript-eslint/ban-types
```

File: src/models/Element.ts [hunk 7]
```diff
@@ -216,46 +216,6 @@ export class NoteElement extends ScoreElement {
     return this._quantitativeNeume;
   }
 
-  public get timeNeume() {
```
Comment (basil) [inline src/models/Element.ts:219]:
```
  259:3  error  All quantitativeNeume signatures should be adjacent     @typescript-eslint/adjacent-overload-signatures
  275:3  error  All timeNeume signatures should be adjacent             @typescript-eslint/adjacent-overload-signatures
  280:3  error  All gorgonNeume signatures should be adjacent           @typescript-eslint/adjacent-overload-signatures
  285:3  error  All secondaryGorgonNeume signatures should be adjacent  @typescript-eslint/adjacent-overload-signatures
  290:3  error  All vocalExpressionNeume signatures should be adjacent  @typescript-eslint/adjacent-overload-signatures
  295:3  error  All accidental signatures should be adjacent            @typescript-eslint/adjacent-overload-signatures
  300:3  error  All secondaryAccidental signatures should be adjacent   @typescript-eslint/adjacent-overload-signatures
  305:3  error  All tertiaryAccidental signatures should be adjacent    @typescript-eslint/adjacent-overload-signatures
  310:3  error  All fthora signatures should be adjacent                @typescript-eslint/adjacent-overload-signatures
  315:3  error  All secondaryFthora signatures should be adjacent       @typescript-eslint/adjacent-overload-signatures
  320:3  error  All tertiaryFthora signatures should be adjacent        @typescript-eslint/adjacent-overload-signatures
```

File: src/utils/observeVisibility.ts [hunk 8]
```diff
@@ -14,10 +14,12 @@ export function processOptions(value: any) {
 }
 
 interface ThrottleOptions {
+  // eslint-disable-next-line @typescript-eslint/ban-types
```
Comment (basil) [inline src/utils/observeVisibility.ts:17]:
```
  17:32  error  Don't use `Function` as a type. The `Function` type accepts any function-like value.
It provides no type safety when calling the function, which can be a common source of bugs.
It also accepts things like class declarations, which will throw at runtime as they will not be called with `new`.
If you are expecting the function to accept certain arguments, you should explicitly define the function shape  @typescript-eslint/ban-types
  21:13  error  Don't use `Function` as a type. The `Function` type accepts any function-like value.
It provides no type safety when calling the function, which can be a common source of bugs.
It also accepts things like class declarations, which will throw at runtime as they will not be called with `new`.
If you are expecting the function to accept certain arguments, you should explicitly define the function shape  @typescript-eslint/ban-types
```

**Review by danielgarthur — APPROVED** [review]

File: package.json (hunk 4 above)
Comment (danielgarthur) [reply package.json:65]: @basil Looks like they just fixed this a few hours ago per the [changelog](https://github.com/prettier/eslint-plugin-prettier/blob/master/CHANGELOG.md#500). I'll complete #163.

**Pull request merged**